src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java

```diff
@@ -931,30 +932,30 @@ private String jsonValueToString(JsonValue jv) {
     }
     
     public List<ControlledVocabularyValue> parseControlledVocabularyValue(DatasetFieldType cvvType, JsonObject json) throws JsonParseException {
+        List<ControlledVocabularyValue> vals = new LinkedList<>();
         try {
             if (cvvType.isAllowMultiples()) {
                 try {
                     json.getJsonArray("value").getValuesAs(JsonObject.class);
                 } catch (ClassCastException cce) {
                     throw new JsonParseException("Invalid values submitted for " + cvvType.getName() + ". It should be an array of values.");
-                }                
-                List<ControlledVocabularyValue> vals = new LinkedList<>();
+                }
                 for (JsonString strVal : json.getJsonArray("value").getValuesAs(JsonString.class)) {
                     String strValue = strVal.getString();
                     ControlledVocabularyValue cvv = datasetFieldSvc.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(cvvType, strValue, lenient);
-                    if (cvv == null) {
+                    if (cvv == null && !allowHarvestingMissingCVV) {
                         throw new ControlledVocabularyException("Value '" + strValue + "' does not exist in type '" + cvvType.getName() + "'", cvvType, strValue);
                     }
-                    // Only add value to the list if it is not a duplicate 
-                    if (strValue.equals("Other")) {
-                        System.out.println("vals = " + vals + ", contains: " + vals.contains(cvv));
-                    }
-                    if (!vals.contains(cvv)) {
-                        vals.add(cvv);
+                    if (cvv != null) {
+                        // Only add value to the list if it is not a duplicate
+                        if (strValue.equals("Other")) {
```

Comment:
Maybe it's time to remove this, or change to logger.fine.

Reply:
Dome